src/section/section_proof_chain.rs

```diff
@@ -6,114 +6,85 @@
 // KIND, either express or implied. Please review the Licences for the specific language governing
 // permissions and limitations relating to use of the SAFE Network Software.
 
-use crate::{
-    error::Result,
-    xor_space::{Prefix, XorName},
-};
-
-#[cfg(all(test, feature = "mock"))]
-use std::iter;
+use std::{collections::HashSet, iter};
 
+/// Block of the section proof chain. Contains the section BLS public key and is signed by the
+/// previous block.
 #[derive(Debug, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 pub struct SectionProofBlock {
-    key_info: SectionKeyInfo,
-    sig: bls::Signature,
+    /// The section key.
+    pub key: bls::PublicKey,
+    /// Signature of the above key, using the previous block.
+    pub signature: bls::Signature,
 }
 
 impl SectionProofBlock {
-    pub fn new(key_info: SectionKeyInfo, sig: bls::Signature) -> Self {
-        Self { key_info, sig }
-    }
-
-    pub fn key_info(&self) -> &SectionKeyInfo {
-        &self.key_info
-    }
-
-    pub fn verify_with_pk(&self, pk: bls::PublicKey) -> bool {
-        if let Ok(to_verify) = self.key_info.serialise_for_signature() {
-            pk.verify(&self.sig, to_verify)
-        } else {
-            false
-        }
-    }
-
-    pub fn prefix(&self) -> &Prefix<XorName> {
-        &self.key_info.prefix
-    }
-
-    pub fn version(&self) -> u64 {
-        self.key_info.version
+    #[cfg_attr(feature = "mock_base", allow(clippy::trivially_copy_pass_by_ref))]
+    pub(crate) fn verify(&self, public_key: &bls::PublicKey) -> bool {
+        public_key.verify(&self.signature, &self.key.to_bytes()[..])
     }
 }
 
 /// Chain of section BLS keys where every key is proven (signed) by the previous key, except the
 /// first one.
 #[derive(Debug, Eq, PartialEq, Clone, Hash, Serialize, Deserialize)]
 pub struct SectionProofChain {
-    head: SectionKeyInfo,
+    head: bls::PublicKey,
     tail: Vec<SectionProofBlock>,
 }
 
 impl SectionProofChain {
     /// Creates new chain consisting of only one block.
-    pub fn new(first: SectionKeyInfo) -> Self {
+    pub fn new(first: bls::PublicKey) -> Self {
         Self {
             head: first,
             tail: Vec::new(),
         }
     }
 
-    pub(crate) fn push(&mut self, block: SectionProofBlock) {
-        if !validate_next_block(self.last_key_info(), &block) {
-            log_or_panic!(
-                log::Level::Error,
-                "Invalid next block: {:?} -> {:?}",
-                self.last_key_info(),
-                block
-            );
-            return;
-        }
-
+    /// Pushes a new block into the chain. No validation is performed.
```

Comment:
wondering why the validation is removed?

Reply:
Mostly because I needed to be able to push invalid blocks for testing purposes. I could have instead created a separate `#[feature = "mock"]` function `push_without_validation` or so, but I decided to keep things simple instead. The validation itself weren't that useful anyway because even when you create an invalid chain, it's going to be rejected by the recipients anyway.

Reply:
hmm...  I thought the `validate` function will be used (though maybe not that useful as you say) for the production code. So, if the removal is due to the testing purpose, I'd prefer make a separate function under the `mock` feature guard.
It's not a strong opinion here, so leave it to you to decide.

Reply:
What I meant is that we don't need to validate the block at the time we are pushing it, because we will validate the whole chain as part of message verification. But I'll add a separate function for it anyway.